### PR TITLE
adding matrix addition

### LINKED ISSUE #34: Matrix addition , multiplication and transpose
## 🚀 Feature

Hi, I am a GSSoC participant I would like to implement this in java, python ,c, c++, and javascript

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

yes

## Pitch

implement matrix addition multiplication and transpose of a matrix

### PR DESCRIPTION
**Pull request by aaadddiii**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### Description

fixes #34 

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

issue #34

### CODE REVIEW

**Comment by aaadddiii** [comment]
@HarshCasper  please review my pr

File: docs/mathematics/matrix-addition.mdx [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+In mathematics, matrix addition is the operation of adding two matrices by adding the corresponding entries together. Two matrices must have an equal number of rows and columns to be added.
+
+## Explanation
+
+Consider two matrices A and B , the sum of two matrices A and B will be a matrix which has the same number of rows and columns as A and B. The sum of A and B, denoted A + B, It is computed by adding corresponding elements of A and B
+ 
+For Example 
+
+![image](https://i.postimg.cc/Jn8yyrzg/image.png)
```
Comment (HarshCasper) [inline docs/mathematics/matrix-addition.mdx:15]: Did you make this diagram?

File: docs/mathematics/matrix-addition.mdx [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+- Input matrix A and matrix B.<br/>
+- Input matrix 1 <br/>
+- Input matrix 2 <br/>
+- If the number of rows and number of columns of matrix A and matrix B is equal<br/>
+- for i=0 to number of rows in A do<br/>
+- for j=0 to number of rows in A do<br/> 
+- Result [i,j] =  A [i,j] + B [i,j]<br/>
+- Display Result
+
```
Comment (HarshCasper) [inline docs/mathematics/matrix-addition.mdx:27]: You don't need to use `<br/>` repeatedly. `mdx` takes care of it

File: docs/mathematics/matrix-addition.mdx [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+title: Matrix Addition
+slug: /mathematics/matrix-addition
+---
+## Introduction
+
+In mathematics, matrix addition is the operation of adding two matrices by adding the corresponding entries together. Two matrices must have an equal number of rows and columns to be added.
+
+## Explanation
+
+Consider two matrices A and B , the sum of two matrices A and B will be a matrix which has the same number of rows and columns as A and B. The sum of A and B, denoted A + B, It is computed by adding corresponding elements of A and B
```
Comment (HarshCasper) [inline docs/mathematics/matrix-addition.mdx:11]: Keep `A` and `B` in backticks

Comment (HarshCasper) [inline docs/mathematics/matrix-addition.mdx:11]: Use proper sentences and grammar. Please do a Grammarly lint at least once.

File: docs/mathematics/matrix-addition.mdx [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+
+## Algorithm
+
+- Input matrix A and matrix B.<br/>
+- Input matrix 1 <br/>
+- Input matrix 2 <br/>
+- If the number of rows and number of columns of matrix A and matrix B is equal<br/>
+- for i=0 to number of rows in A do<br/>
+- for j=0 to number of rows in A do<br/> 
+- Result [i,j] =  A [i,j] + B [i,j]<br/>
```
Comment (HarshCasper) [inline docs/mathematics/matrix-addition.mdx:25]: Keep `i`, `j`, `A`, `B` in backticks

File: docs/mathematics/matrix-addition.mdx [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+The resultant matrix is:
+5 7 10
+13 9 10
+9 9 14
+```
+
+## Complexity Analysis
+
+- Time complexity - O(n * m)
+- Space complexity - O(n*m),, where `n` and `m` are the number of rows and columns respectively
```
Comment (HarshCasper) [inline docs/mathematics/matrix-addition.mdx:172]: Keep the values in backticks

File: docs/mathematics/matrix-addition.mdx [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+1 2 4
+5 3 6
+7 5 8
+Enter the values of the second matrix
+4 5 6
+8 6 4
+2 4 6
+```
+
+### Output
```
Comment (HarshCasper) [inline docs/mathematics/matrix-addition.mdx:160]: Use Heading-4

File: docs/mathematics/matrix-addition.mdx [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+    print()
+    
+```
+
+</TabItem>
+</Tabs>
+
+## Sample Input and Output
+
+### Input
```
Comment (HarshCasper) [inline docs/mathematics/matrix-addition.mdx:145]: Use Heading-4

File: docs/mathematics/matrix-addition.mdx [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+- If the number of rows and number of columns of matrix A and matrix B is equal<br/>
+- for i=0 to number of rows in A do<br/>
+- for j=0 to number of rows in A do<br/> 
+- Result [i,j] =  A [i,j] + B [i,j]<br/>
+- Display Result
+
+```
+for i <-- 0 to row_a do
+    for j <-- 0 to col_a do
+        result[i][j]  = A[i][j] + B[i][j]
```
Comment (HarshCasper) [inline docs/mathematics/matrix-addition.mdx:31]: Use an example/explanation. Not pseudocode

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: docs/mathematics/matrix-addition.mdx (hunk 1 above)
Comment (aaadddiii) [reply docs/mathematics/matrix-addition.mdx:15]: Yes

**Comment by aaadddiii** [comment]
@HarshCasper  I have made the changes , can you please review my pr

File: docs/mathematics/matrix-addition.mdx [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+## Algorithm
+
+- Input matrix `A` and matrix `B`
+- Input matrix `A`
+- Input matrix `B` 
+- If the number of rows and number of columns of matrix `A` and matrix `B` is equal
+- loop from `i`=0 to number of rows in A do
+- loop from `j`=0 to number of rows in A do 
+- `Result[i,j]` =  `A[i,j]` + `B[i,j]`
+- Display `Result`
```
Comment (HarshCasper) [inline docs/mathematics/matrix-addition.mdx:26]: Use text for algorithm steps. Use formulae to complement them

File: docs/mathematics/matrix-addition.mdx [hunk 10]
```diff
@@ -0,0 +1,172 @@
...
+}>
+
+<TabItem value="cpp">
+
+```cpp
+// C++ program to implement Matrix Addition
+#include <bits/stdc++.h>
+using namespace std;
+
+int main()
```
Comment (HarshCasper) [inline docs/mathematics/matrix-addition.mdx:49]: Don't put the entire logic in the `main()` function. Use `main()` only for the driver code.

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by aaadddiii** [comment]
@HarshCasper  I have made the changes please review it

File: docs/mathematics/matrix-addition.mdx [hunk 11]
```diff
@@ -0,0 +1,183 @@
...
+## Explanation
+
+Consider two matrices `A` and `B`, the sum of two matrices `A` and `B` will be a matrix that has the same number of rows and columns as `A` and `B`. The sum of `A` and `B` denoted `A` + `B`, is computed by adding corresponding elements of `A` and `B`
+ 
+For Example 
+
+![image](https://i.postimg.cc/Jn8yyrzg/image.png)
+
+## Algorithm
+
```
Comment (vybhav72954) [inline docs/mathematics/matrix-addition.mdx:18]: The algorithm can be much more plain text-based, please make changes accordingly.

**Review by vybhav72954 — CHANGES_REQUESTED** [review]

**Comment by aaadddiii** [comment]
@vybhav72954  I have made the changes, please review it

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**